src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -3286,25 +3285,107 @@ void MavlinkReceiver::update_rx_stats(const mavlink_message_t &message)
 		PX4_WARN("Max remote components of %u used up", MAX_REMOTE_COMPONENTS);
 		_warned_component_states_full_once = true;
 	}
+
+
+	if (_message_statistics_enabled) {
+		if (_received_msg_stats == nullptr) {
+			_received_msg_stats = new ReceivedMessageStats[MAX_MSG_STAT_SLOTS];
```

Comment:
If `new` can't allocate, is it a problem if we keep calling it in every iteration?

Reply:
I think it's relatively harmless, or at least less bad than tracking more state.